components/renku_data_services/connected_services/db.py

```diff
@@ -461,13 +461,20 @@ async def get_oauth2_app_installations(
             connection,
             adapter,
         ):
-            # NOTE: App installations are only available from GitHub
-            if connection.client.kind == models.ProviderKind.github and isinstance(adapter, GitHubAdapter):
+            # NOTE: App installations are only available from GitHub when using a "GitHub App"
+            #       it doesn't work for "OAuth App". Currently we guess handling a "Github App"
+            #       if no image_registry_url is specified
+            if (
+                connection.client.kind == models.ProviderKind.github
+                and connection.client.url == "https://github.com"
```

Comment:
This will break for GitHub Enterprise (the URL of the GitHub instance is customized for GitHub Enterprise).

I think it is more robust solution here is to test again `image_registry_url`.

We have one GitHub Enterprise integration working in production:
<img width="852" height="330" alt="Image" src="https://github.com/user-attachments/assets/17b0d794-8195-4563-b591-be0f9af7b514" />
```json
{
        "id": "github.zhaw.ch",
        "kind": "github",
        "app_slug": "zhaw-github",
        "client_id": "redacted",
        "client_secret": "redacted",
        "display_name": "ZHAW GitHub",
        "scope": "",
        "url": "https://github.zhaw.ch",
        "use_pkce": false
    }
```